ddp-workspace/projects/ddp-dsm-ui/src/app/stool-upload/stool-upload.component.ts

```diff
@@ -0,0 +1,192 @@
+import { Result } from '../utils/result.model';
+import { RoleService } from '../services/role.service';
+
+@Component({
+  selector: 'app-stool-upload',
+  templateUrl: './stool-upload.component.html',
+  styleUrls: ['./stool-upload.component.css']
+})
+
+export class StoolUploadComponent implements OnInit {
+  @ViewChild(ModalComponent)
+  public modal: ModalComponent;
+
+  @ViewChild(FieldFilepickerComponent)
+  public filepicker: FieldFilepickerComponent;
+
+  errorMessage: string;
+  additionalMessage: string;
+
+  loading = false;
+  uploadPossible = false;
+
+  kitTypes: Array<KitType> = [];
+  uploadReasons: Array<string> = [];
+  carriers: Array<string> = [];
+  kitType: KitType;
+
+  file: File = null;
+
+  failedParticipants: Array<UploadParticipant> = [];
+  duplicateParticipants: Array<UploadParticipant> = [];
+  specialKits: Array<UploadParticipant> = [];
+
+  realmNameStoredForFile: string;
+  allowedToSeeInformation = false;
+  specialMessage: string;
+
+  constructor( private dsmService: DSMService, private auth: Auth, private compService: ComponentService, private route: ActivatedRoute,
+               private role: RoleService ) {
+    if (!auth.authenticated()) {
+      auth.logout();
+    }
+    this.realmNameStoredForFile = localStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+    this.route.queryParams.subscribe(params => {
+      const realm = params[ DSMService.REALM ] || null;
+      if (realm != null && realm !== '') {
+        //        this.compService.realmMenu = realm;
+        this.checkRight();
+      }
+    });
+  }
+
+  private checkRight(): void {
+    this.allowedToSeeInformation = false;
+    this.additionalMessage = null;
+    let jsonData: any[];
+    this.dsmService.getRealmsAllowed(Statics.SHIPPING).subscribe({
+      next: data => {
+        jsonData = data;
+        jsonData.forEach((val) => {
+          if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) === val) {
+            this.allowedToSeeInformation = true;
+            this.getPossibleKitType();
+          }
+        });
+        if (!this.allowedToSeeInformation) {
+          this.additionalMessage = 'You are not allowed to see information of the selected study at that category';
+        }
+      },
+      error: () => null
+    });
+  }
+
+  ngOnInit(): void {
+    if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null) {
+      this.checkRight();
+    } else {
+      this.additionalMessage = 'Please select a study';
+    }
+    window.scrollTo(0, 0);
+  }
+
+  getPossibleKitType(): void {
+    if (this.isSelectedRealm()) {
+      this.loading = true;
+      let jsonData: any[];
+      this.dsmService.getKitTypes(localStorage.getItem(ComponentService.MENU_SELECTED_REALM)).subscribe({
+        next: data => {
+          this.kitTypes = [];
+          jsonData = data;
+          jsonData.forEach((val) => {
+            const kitType = KitType.parse(val);
+            if(kitType.name =='STOOL'){
```

Comment:
Please move the "STOOL" string into the const variable, or even better, move it into the enums file;